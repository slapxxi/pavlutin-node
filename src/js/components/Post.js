import React from 'react';
import Icon from 'react-fontawesome';
import Markdown from 'react-markdown';
import DisqusThread from 'react-disqus-thread';
import { toHumanReadableDate } from '../utils';

function Post({ post }) {
  const URL = `http://slavapavlutin.com/blog/${post.slug}`;
  return (
    <article className="post">
      <h1 className="post__title">{post.title}</h1>
      <div className="post__meta">
        <Icon name="clock-o" /> {' '}
        {toHumanReadableDate(post.createdAt)}
      </div>
      { post.img ? (
        <div className="post__image">
          <img src={`/img/${post.img}`} alt={`${post.title}`} />
        </div>
        ) : null
      }
      <div className="post__description">
        <Markdown source={post.description || ''} />
      </div>
      <div className="post__content">
        <Markdown source={post.content.replace(/\\n/g, '\n')} />
      </div>
      <DisqusThread
        identifier={post.slug + post.id}
        title={post.title}
        url={URL}
        category_id={post.id}
        // shortname="slavapavlutin-com"
      />
    </article>
  );
}

export default Post;
