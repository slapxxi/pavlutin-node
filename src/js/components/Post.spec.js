import React from 'react';
import { mount, shallow } from 'enzyme';
import Post from './Post';
import { singleLine } from '../../../test/test-helper';

const post = {
  id: 0,
  title: 'Test Post',
  content: '**content**',
  description: 'empty',
  author: 'bot',
  slug: 'test-post',
  tags: ['test', 'react'],
};

it('renders post title', () => {
  const result = shallow(<Post post={post} />);
  expect(result.find('.post__title').text()).toBe('Test Post');
});

it('renders post content', () => {
  const result = mount(<Post post={post} />);
  expect(result.find('.post__content').text()).toBe('content');
});

it('converts post content from Markdown', () => {
  const result = shallow(<Post post={post} />);
  const expected = singleLine`
    <div class="post__content">
      <div>
        <p><strong>content</strong></p>
      </div>
    </div>
  `;
  expect(result.find('.post__content').html()).toBe(expected);
});

it('renders post metadata', () => {
  const result = mount(<Post post={post} />);
  expect(result.find('.post__meta').text()).toBe('  a few seconds ago');
});

it('renders post image', () => {
  const postWithImage = { ...post, img: 'imagetoBepng' };
  const result = shallow(<Post post={postWithImage} />);
  expect(result.find('.post__image').length).toBe(1);
});

it('renders no image if missing', () => {
  const result = mount(<Post post={post} />);
  expect(result.find('.post__image').length).toBe(0);
});
