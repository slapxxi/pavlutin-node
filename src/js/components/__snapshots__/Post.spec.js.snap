// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders post 1`] = `
<article
  className="post"
>
  <h1
    className="post__title"
  >
    Test Post
  </h1>
  <div
    className="post__meta"
  >
    <span
      aria-hidden={true}
      className="fa fa-clock-o"
    />
     
     
    a few seconds ago
  </div>
  <div
    className="post__description"
  >
    <div>
      <p>
        empty
      </p>
    </div>
  </div>
  <div
    className="post__content"
  >
    <div>
      <p>
        <strong>
          content
        </strong>
      </p>
    </div>
  </div>
</article>
`;

exports[`renders post image 1`] = `
<article
  className="post"
>
  <h1
    className="post__title"
  >
    Test Post
  </h1>
  <div
    className="post__meta"
  >
    <span
      aria-hidden={true}
      className="fa fa-clock-o"
    />
     
     
    a few seconds ago
  </div>
  <div
    className="post__image"
  >
    <img
      alt="Test Post"
      src="/img/test.png"
    />
  </div>
  <div
    className="post__description"
  >
    <div>
      <p>
        empty
      </p>
    </div>
  </div>
  <div
    className="post__content"
  >
    <div>
      <p>
        <strong>
          content
        </strong>
      </p>
    </div>
  </div>
</article>
`;
